wfl/configset.py

```diff
@@ -324,7 +324,7 @@ def __str__(self):
         return s
 
 
-class ConfigSet_out:
+class OutputSpec:
     """Thin output layer for configurations into files, ABCD, or atomic configs
 
```

Comment:
I find the `all_or_none=True, force=True` combination for skipping operations that would overwrite the `OutputSpec` files a bit cryptic. Would it be worth adding a tag `skip_if_complete` (or similar) in addition or instead?

Reply:
The problem is that the actual skipping functionality is in `autoparallelize`, not really a part of the output spec.   It would definitely be slightly more sensible if `force=True` wasn't necessary for the skipping.  

Anyway, I agree that putting those in is a bit obscure, but isn't that the default behavior now anyway, so users don't really have to worry about it?

Reply:
OK.  I stand by my earlier statement - the actual skipping functionality is in `autoparallelize()`, and it just requires an `OutputSpec` setting that makes it safe.  I also stand (mostly) by my initial claim that the `OutputSpec` arguments communicate what it's going to do, but it's just that it's not obvious that those behaviors are required to get safe skipping behavior.  The "mostly" is because I do agree that it'd be better if `force` didn't matter for the skipping.  I might be able to fix that, if we think it's worth it.

However, for what it's worth, you **do not** have to pass force=True.  If you pass `all_or_none=True` (which is actually the default now) and **no** `force`, it defaults to `force=True`, and the skipping works.

Reply:
okay! I like that `all_or_none=True` (the default) sets the `force` as well. 

I think "all_or_none" came from slightly different reasoning than "skip if the file is present", but I think that's going to be the more frequent use./? What do you think about renaming "all_or_none" to reflect that and be slightly more self-explanatory?

Reply:
If we can come up with a better name, great, but since `OutputSpec` is not the same as `autoparallelize`, and it's `autoparallelize` that does the skipping, I don't think the "skipping" part makes sense to be in the `OutputSpec` arguments.  The standard name for this behavior (an operation that's all or none) is an atomic operation, so a more technically correct (if not necessarily more clear) name for that argument could be `atomic_write`, but I'm not sure that really helps you.   The fundamental problem is that there's nothing inherent to the `OutputSpec` that makes it skip - it's only `autoparallelize` that uses certain `OutputSpec` _behavior_.

Reply:
okay, I don't have any better ideas, let's leave it as is.